src/daft-logical-plan/src/builder.rs

```diff
@@ -506,6 +506,13 @@ impl LogicalPlanBuilder {
         Ok(self.with_new_plan(logical_plan))
     }
 
+    pub fn intersect(&self, other: &Self, is_all: bool) -> DaftResult<Self> {
+        let logical_plan: LogicalPlan =
+            ops::Intersect::try_new(self.plan.clone(), other.plan.clone(), is_all)?
+                .to_optimized_join()?;
+        Ok(self.with_new_plan(logical_plan))
+    }
```

Comment:
instead of immediately converting it to a join, I think we should instead defer this until `translate` stage. 

If we want to have a concept of a logical `Intersect`, then I don't think we should immediately convert it to a logical join, but only convert it once to a physical join during the logical -> physical translation

Reply:
ok, Lets go with what's in here now. We can always revisit it needed 😅